zerver/views/message_edit.py

```diff
@@ -93,13 +93,17 @@ def get_message_edit_history(
     user_profile: UserProfile,
     message_id: int = REQ(converter=to_non_negative_int, path_only=True),
 ) -> HttpResponse:
-    if not user_profile.realm.allow_edit_history:
+    if user_profile.realm.show_message_edit_history == 3:
         raise JsonableError(_("Message edit history is disabled in this organization"))
     message, ignored_user_message = access_message(user_profile, message_id)
 
     # Extract the message edit history from the message
     if message.edit_history is not None:
         raw_edit_history = orjson.loads(message.edit_history)
+        if user_profile.realm.show_message_edit_history == 2:
+            for history in raw_edit_history[:]:
+                if "prev_content" in history:
+                    raw_edit_history.remove(history)
```

Comment:
Let's make this block a function, perhaps `visible_edit_history_for_message(message, realm)`; we can put it in `zerver/lib/message.py` and use it here as well as other code paths that need it.

Reply:
This is a good way to go, on it!